pkg/operator/controller/controllerregistrar/reconciler.go

```diff
@@ -34,19 +36,22 @@ type Controller struct {
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
-	if r.allControllersAdded() {
-		return reconcile.Result{}, nil
-	}
-
 	garden := &operatorv1alpha1.Garden{}
 	if err := r.Manager.GetClient().Get(ctx, request.NamespacedName, garden); err != nil {
 		if apierrors.IsNotFound(err) {
-			log.V(1).Info("Object is gone, stop reconciling")
+			// Shut down Gardener-Operator in case the garden was deleted.
+			// This is a pragmatic way to deregister all controllers that depend on the existence of a garden cluster.
+			log.Info("Terminating Gardener-Operator after garden deletion")
```

Comment:
```suggestion
			log.Info("Terminating gardener-operator after garden deletion")
```

Nit :)

Reply:
Inspired by https://github.com/gardener/gardener/blob/6daf93f73fe7e773ea8deb436ae76f52e6febd89/pkg/gardenlet/bootstrap/certificate/certificate_rotation.go#L108 😱 😱 😱 (still don't get the intended writing style here 🤣)

Reply:
:D Let's have another session in the office to discuss about it !